src/main/kotlin/no/skatteetaten/aurora/cantus/controller/DockerRegistryController.kt

```diff
@@ -18,14 +20,44 @@ import org.springframework.web.bind.annotation.RestController
 
 data class TagUrlsWrapper(val tagUrls: List<String>)
 
+private val logger = KotlinLogging.logger {}
+
 @RestController
 class DockerRegistryController(
     val dockerRegistryService: DockerRegistryService,
     val imageTagResourceAssembler: ImageTagResourceAssembler,
     val imageRepoCommandAssembler: ImageRepoCommandAssembler,
-    val threadPoolContext: ExecutorCoroutineDispatcher
+    val threadPoolContext: ExecutorCoroutineDispatcher,
+    val aurora: AuroraIntegration
 ) {
 
+    /*
+      TODO: For now the bearer token is only for the push registry, we need to create a composite token in the future if pull demands authroization
+     */
+    @PostMapping("/tag")
+    fun tagDockerImage(
+        @RequestBody tagCommand: TagCommand,
+        @RequestHeader(required = true, value = HttpHeaders.AUTHORIZATION) bearerToken: String
+    ): AuroraResponse<TagCommandResource> {
+
+        val from = imageRepoCommandAssembler.createAndValidateCommand(tagCommand.from)
+        val to = imageRepoCommandAssembler.createAndValidateCommand(tagCommand.to, bearerToken)
+
+        return try {
+            if (from.imageTag == null) throw BadRequestException("From spec=${tagCommand.from} does not contain a tag")
+            if (to.imageTag == null) throw BadRequestException("To spec=${tagCommand.from} does not contain a tag")
+
+            val result = dockerRegistryService.tagImage(from, to)
+            AuroraResponse(
+                success = true,
+                message = "${from.fullRepoCommand} -> ${to.fullRepoCommand}",
+                items = listOf(TagCommandResource(result))
+            )
+        } catch (e: Exception) {
```

Comment:
for generisk å fange `Exception`? vil vi at alle typer exception sendes inn her?

Reply:
I cantus har vi ikke noen ControllerAdvice som tilpasser exceptions så de har samme konvolutt som resten av koden. Derfor catches det veldig bredt her. VI kan la være å catcher her hvis vi innfører en controllerAdvice som gjør det over.

Reply:
hmm, skjønner. usikker på hva jeg foretrekker egentlig...